scripts/enumerable.rb

```diff
@@ -0,0 +1,246 @@
+# frozen_string_literal: true
+
+module Enumerable
+  def my_each
+    return return_enum unless block_given?
+
+    i = 0
+    while i < size
+      yield(self[i])
+      i += 1
+    end
+  end
+
+  def my_each_with_index
+    return return_enum unless block_given?
+
+    i = 0
+    while i < size
+      yield(self[i], i)
+      i += 1
+    end
+  end
+
+  def my_select
+    return return_enum unless block_given?
+
+    select = []
+    my_each { |item| select << item if yield item }
+    select
+  end
+
+  def my_all?(pattern = nil)
+    all = true
+
+# rubocop:disable all
+    if block_given?
+      my_each { |item| all = false unless yield item }
+    elsif pattern
+      my_each { |item| all = false unless pattern === item }
+    else
+      my_each { |item| all = false unless item }
+    end
+
+# rubocop:enable all
+    all
+  end
+
+  def my_any?(pattern = nil)
+    any = false
+
+# rubocop:disable all
+    if block_given?
+      my_each { |item| any = true if yield item }
+    elsif pattern
+      my_each { |item| any = true if pattern === item}
+    else
+      my_each { |item| any = true if item }
+    end
+
+# rubocop:enable all
+    any
+  end
+
+  def my_none?(pattern = nil)
+    none = true
+
+# rubocop:disable all
+    if block_given?
+      my_each { |item| none = false if yield item }
+    elsif pattern
+      my_each { |item| none = false if pattern === item}
+    else
+      my_each { |item| none = false if item }
+    end
+
+# rubocop:enable all
+    none
+  end
+
+  def my_count(compare = nil)
+    count = 0
+    if block_given?
+      my_each { |item| count += 1 if yield item }
+    elsif !compare.nil?
+      my_each { |item| count += 1 if item == compare }
+    else
+      each { |_item| count += 1 }
+    end
+    count
+  end
+
+  def my_map(my_proc = nil)
+    map = []
+    if my_proc
+      my_proc.call
+    elsif block_given?
+      each { |item| map << yield(item) }
+    else
+      return to_enum
+    end
+    map
+  end
+
+  def my_inject(sym = nil, memo = self[0])
+    raise 'No block given' unless block_given? || !sym.nil?
+
+# rubocop:disable all
+    if sym
+      my_each { |item|
+        next if item===self[0]
+        memo=memo.send(sym,item)
+      }
+    else
+      my_each { |item|
+        next if item===self[0]
+        memo = yield(memo, item)
+      }
+    end
+# rubocop:enable all
+
+    memo
+  end
+
+  def multiply_els(array)
+    array.my_inject { |mult, item| mult * item }
+  end
+
```

Comment:
Please, let's move `multiply_els` outside the Module. We should be able to use  `multiply_els` like this `multiply_els([2, 4, 5])` not like this  `[2, 4, 5].multiply_els([2, 4, 5])`

Reply:
I am sorry I missed this in the previous review even though it was on the main summary of the review. :man_facepalming: